### PR TITLE
Allow enabling managed Prometheus in beta cluster submodules

### LINKED ISSUE #1290: missing `enable-managed-prometheus` for GKE cluster
### TL;DR

GKE setting:
https://cloud.google.com/stackdriver/docs/managed-prometheus/setup-managed#config-mgd-collection

kubectl apply -f https://raw.githubusercontent.com/GoogleCloudPlatform/prometheus-engine/v0.4.1/manifests/setup.yaml

kubectl apply -f https://raw.githubusercontent.com/GoogleCloudPlatform/prometheus-engine/v0.4.1/manifests/operator.yaml

but, did not find configuration from terraform config.

### Terraform Resources

_No response_

### Detailed design

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by jroiseux**
Managed Prometheus can now be enabled with the release of the beta provider, `4.25.0`.

Adds an input, `monitoring_enable_managed_prometheus`, to beta cluster submodules using a nested dynamic block in the pre-existing `monitoring_config` block.

The `monitoring_config` block previously only accepted the `monitoring_enabled_components` input, which is now set to null if it is not specified, in order to prevent `SYSTEM_COMPONENTS` from being disabled when `monitoring_enable_managed_prometheus` is set to true.

Fixes PR #1298 

Resolves #1290 and #1167

### CODE REVIEW

**Comment by comment-bot-dev** [comment]
@jroiseux
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

File: autogen/main/cluster.tf.tmpl [hunk 1]
```diff
@@ -89,10 +89,18 @@ resource "google_container_cluster" "primary" {
...
-      enable_components = var.monitoring_enabled_components
+      enable_components = length(var.monitoring_enabled_components) > 0 ? var.monitoring_enabled_components : null
+
+      dynamic "managed_prometheus" {
+        for_each = var.monitoring_enable_managed_prometheus ? [1] : []
+
+        content {
+          enabled = var.monitoring_enable_managed_prometheus
+        }
+      }
```
Comment (ykzts) [inline autogen/main/cluster.tf.tmpl:103]: You can also enable `managed_prometheus` if `autopilot_cluster` is `true` (but Autopilot cannot enable `enable_components`).

Comment (bharathkkb) [reply autogen/main/cluster.tf.tmpl:103]: @ykzts Can you open an issue so we can track this for autopilot support?

**Comment by cpanato** [comment]
any plans to release this feature?

**Pull request merged**